frontend/src/app/shared/components/principal/principal-helper.ts

```diff
@@ -26,10 +26,40 @@
 // See COPYRIGHT and LICENSE files for more details.
 //++
 
+import { PrincipalLike } from 'core-app/shared/components/principal/principal-types';
+import { Principal } from 'core-app/core/state/principals/principal.model';
+
 export namespace PrincipalHelper {
   export type PrincipalType = 'user'|'placeholder_user'|'group';
   export type PrincipalPluralType = 'users'|'placeholder_users'|'groups';
 
+  /*
+   * This function is a helper that wraps around the old HalResource based principal type and the new interface based one.
+   * TypeScript constantly screams that the href member does not exist (which I know, that's why the check is there you dumbo)
+   * In order to minimize the `@ts-ignore` pollution around the codebase it's centralized here.
+   *
+   * TODO: Remove old HalResource stuff :P
+   */
+  export function hrefFromPrincipal(p:Principal|PrincipalLike):string {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (p.href) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      return p.href;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    if (p._links) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      return (p._links as Record<string, string>).self.href as string;
+    }
+
+    return '';
+  }
+
```

Comment:
The `@ts-ignore` here is shadowing the issue that the href can be undefined in case you pass a PrincipalLike. Please try to avoid ts-ignore if at all possible, even if it results in more painful type guards.

I've add a `|| ''` guard to ensure the function definition does not change, but the function return type should actually be `string|undefined`. That is easier to deal with on the consumer side than an empty string I reckon.

```suggestion
  /*
   * This function is a helper that wraps around the old HalResource based principal type and the new interface based one.
   *
   * TODO: Remove old HalResource stuff :P
   */
  export function hrefFromPrincipal(p:Principal|PrincipalLike):string {
    if ((p as PrincipalLike).href) {
      return (p as PrincipalLike).href || '';
    }

    if ((p as Principal)._links) {
      const self = (p as Principal)._links.self as HalSourceLink;
      return self.href || '';
    }

    return '';
  }
```